drivers/adc/adc_stm32.c

```diff
@@ -277,6 +277,35 @@ static void adc_stm32_start_conversion(const struct device *dev)
 #endif
 }
 
+#if !defined(CONFIG_SOC_SERIES_STM32F2X) && \
+	!defined(CONFIG_SOC_SERIES_STM32F4X) && \
+	!defined(CONFIG_SOC_SERIES_STM32F7X) && \
+	!defined(CONFIG_SOC_SERIES_STM32F1X) && \
+	!defined(CONFIG_SOC_SERIES_STM32L1X)
+static void adc_stm32_calib(const struct device *dev)
+{
+	const struct adc_stm32_cfg *config =
+		(const struct adc_stm32_cfg *)dev->config;
+	ADC_TypeDef *adc = config->base;
+
+#if defined(CONFIG_SOC_SERIES_STM32F3X) || \
+	defined(CONFIG_SOC_SERIES_STM32L4X) || \
+	defined(CONFIG_SOC_SERIES_STM32WBX) || \
+	defined(CONFIG_SOC_SERIES_STM32G4X)
+	LL_ADC_StartCalibration(adc, LL_ADC_SINGLE_ENDED);
+#elif defined(CONFIG_SOC_SERIES_STM32F0X) || \
+	defined(CONFIG_SOC_SERIES_STM32G0X) || \
+	defined(CONFIG_SOC_SERIES_STM32L0X) || \
+	defined(CONFIG_SOC_SERIES_STM32WLX)
+	LL_ADC_StartCalibration(adc);
+#elif defined(CONFIG_SOC_SERIES_STM32H7X)
+	LL_ADC_StartCalibration(adc, LL_ADC_CALIB_OFFSET, LL_ADC_SINGLE_ENDED);
+#endif
```

Comment:
What about  the following ?
```
      LL_ADC_StartCalibration(adc,
#if defined(CONFIG_SOC_SERIES_STM32H7X)
                                           LL_ADC_CALIB_OFFSET
#endif
#if defined(CONFIG_SOC_SERIES_STM32F0X) || \
     defined(CONFIG_SOC_SERIES_STM32G0X) || \
     defined(CONFIG_SOC_SERIES_STM32H7X) || \
     defined(CONFIG_SOC_SERIES_STM32L0X) || \
     defined(CONFIG_SOC_SERIES_STM32WLX)
                                           LL_ADC_SINGLE_ENDED
#endif
                                           );
```

Reply:
Personally I don't find this easier to read and maintain, I prefer the previous implementation
By the way it seems it is a source of mistake :smile: :
* "," is misplaced between "adc" and  LL_ADC_CALIB_OFFSET, it should be in the #if
* there is a missing "," between both LL_xxx parameters)

Reply:
I agree with @ABOSTM. I find it easier to debug seeing the full call instead of having to piece it together from some ifdefs.